usb.c

```diff
@@ -1200,13 +1200,70 @@ static int usb_fill_context_info(struct iio_context_info *info,
 	return 0;
 }
 
-int usb_context_scan(struct iio_scan_result *scan_result)
+#define SCAN_WILDCARD "*"
+#define SCAN_DELIMITER ','
+
+static int parse_value(const char *str)
+{
+	uint32_t val;
+	char *ptr;
+
+	/* if it is null or '*' accept any */
+	if (str == NULL || !strncmp(str, SCAN_WILDCARD, sizeof(SCAN_WILDCARD)))
+		return 0;
+	/* if it's empty, that's not valid */
+	if (*str == '\0')
+		return -1;
+
+	val = strtoul(str, &ptr, 16);
+
+	if (ptr == str)
+		return -1;
+
+	/* should be either ":" or end of string */
+	if (*ptr != SCAN_DELIMITER && *ptr != '\0')
```

Comment:
Since you now only get a `vid:pid` string, everything else than a `\0` would be an error here.

Reply:
the string can be `vid:` or `vid:*` or `vid:pid`

Reply:
I don't think we should allow `vid:`. Maybe pass a boolean to the function that tells you whether you're trying to get the VID or PID. If it's true then issue an error if the character is not SCAN_DELIMITER, if it's false issue an error if the character is not \0.